commit 5c66d6f066c0b7427ed44b0d1a01006857d1f3d2
Author: Johan Hedberg <johan.hedberg@nokia.com>
Date:   Fri Sep 17 10:23:08 2010 +0300

    Check for agent in session_free before calling agent_cancel
    
    If the agent existed during the session the agent pointer will be NULL
    and agent_cancel shouldn't be called.

diff --git a/src/adapter.c b/src/adapter.c
index fd4cf99..670b8bd 100644
--- a/src/adapter.c
+++ b/src/adapter.c
@@ -757,7 +757,7 @@ static void session_free(struct session_req *req)
 
 	if (req->msg) {
 		dbus_message_unref(req->msg);
-		if (req->mode)
+		if (req->mode && req->adapter->agent)
 			agent_cancel(req->adapter->agent);
 	}
 
